lib/proxysql_gtid.cpp

```diff
@@ -0,0 +1,116 @@
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#include <string>
+
+#include "proxysql_gtid.h"
+
+
+// Initializes a GTID interval.
+// Implemented via private method as C++03 does not support delegated constructors :(
+void Gtid_Interval::init(const int64_t _start, const int64_t _end) {
```

Comment:
You can use constructor delegation. ProxySQL compilation requires `C++11` or above ([ref](https://github.com/sysown/proxysql/blob/5a314d23648e69bea56f78c47085a30799f17169/Makefile#L52)).